Desktop/components/JASP/Widgets/CustomMenu.qml

```diff
@@ -43,6 +43,74 @@ FocusScope
 	property var    sourceItem  : null
 	property point	scrollOri	: "0,0" //Just for other qmls to use as a general storage of the origin of their scrolling
 
+    property int    currentIndex: -1
+
+    Keys.onPressed: {
+
+        let analysisName    = "";
+        let nextIndex       = -1;
+        let moveToNextIndex = false;
+
+        if (event.key === Qt.Key_Up) {
+
+            if (menu.currentIndex > 0) {
+
+                // There should be a better way to do this!
```

Comment:
Yes. I think it is better to handle this in the model self. Here you are presuming that this is the menu for the analyses, but the CustomMenu is used in other situation. So let the model decide what to do with up and down action.

Reply:
Or you could add in the props an extra option (like functionCall): upCall, downCall. If they are not present, then a default up and down action can be defined.